src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -738,7 +929,8 @@ private int getNumberOfResponses(
         return numChoicesSelected;
     }
 
-    private int getNumberOfNonEmptyResponsesOfQuestion(List<String> answerStrings, Map<String, Integer> answerFrequency) {
+    private static int getNumberOfNonEmptyResponsesOfQuestion(List<String> answerStrings, Map<String,
```

Comment:
When will the responses be empty?

Reply:
I am not sure about that. This was the existing implementation. The method is checking whether any of the answer String is empty in the list of answer strings, and if it is, we are not counting it. But I am not sure how we are going to get the empty answer, until we are submitting the answers in an action test?

I guess this is just an extra layer of protection?

Reply:
A quick git blame tells me it is the legacy code we left for `None of the above`choice in MSQ question. https://github.com/TEAMMATES/teammates/commit/28f8a6699b22bbb4923e4562c245379c60cf2251

@damithc It seems that we used to supoort `None of the above` option and use an empty string as an answer for the options (which results in the code above). And in https://github.com/TEAMMATES/teammates/pull/4002, we migrate to use `other answer` option. Has the empty data been migrated to use the `other` answer options at that time?

Reply:
> Has the empty data been migrated to use the other answer options at that time?

Not sure actually. I don't think this is possible to check using a GQL query via the datastore admin dashboard. May be rig up a client script to check if the old data has been migrated?

On a related note, in future cases of this nature, we should create issues for step 2 and 3 of the data migration and cross refer them from step 1 PR. That way, we can keep track of whether the data migration was done.

Reply:
Yes, I would like to try.

Reply:
@damithc @xpdavid @bqnguyen94 

>It seems that we used to supoort None of the above option

I just realised that writing the data migration script is not necessary as we **still** support `None of the above` option. Below are screenshots of situations where the option is supported. 

**When `Other` option is not selected, typical case**
![screenshot 544](https://user-images.githubusercontent.com/10681755/42367914-400b669c-8123-11e8-8ead-b5a1ad2de2ec.png)

**When `Other` option is selected**
![screenshot 545](https://user-images.githubusercontent.com/10681755/42367923-461e63c2-8123-11e8-860e-383bb415ccf5.png)

**When `Maximum number of options a respondent is allowed to select` is selected** 
![screenshot 546](https://user-images.githubusercontent.com/10681755/42367933-5514ea5e-8123-11e8-9822-fd1d8c2eaf90.png)

**Only when `Minimum number of options a respondent is allowed to select` option is selected, we don't show `None of the above` option.**
![screenshot 547](https://user-images.githubusercontent.com/10681755/42368079-c0531a3e-8123-11e8-817c-3bea2960b763.png)

>And in #4002, we migrate to use other answer option. Has the empty data been migrated to use the other answer options at that time?

I guess we don't have a record of the migration because we did not replace the `None of the above` option with `Other` option, we just added the `Other` option.

Reply:
I suspect you are right. In fact, the purpose of the `None of the above` is to indicate that the question is being answered but no options are selected; the `Other` option has a different purpose.

Reply:
Thank you for your investigation! Yes, we don't need to do anything.

Reply:
@sukanta-27 On a separate note, can you update the document for the MSQ question to indicate that we store none of the above as empty string? Both of us get confused by the empty string checking in the stats generation.

Reply:
Yes, good point. Will add a comment.